hack/verify-golangci-lint.sh

```diff
@@ -76,9 +124,14 @@ if [ "$res" -eq 0 ]; then
 else
   {
     echo
-    echo 'Please review the above warnings. You can test via "./hack/verify-golangci-lint.sh"'
+    echo "Please review the above warnings. You can test via \"${invocation[*]}\""
     echo 'If the above warnings do not make sense, you can exempt this warning with a comment'
     echo ' (if your reviewer is okay with it).'
+    if [ "$strict" ]; then
+        echo 'The more strict .golangci-strict.yaml was used. If you feel that this warns about issues'
+        echo 'that should be ignored by default, then please discuss with your reviewer and propose'
+        echo 'a change for .golangci-strict.yaml as part of your PR.'
```

Comment:
This is up for debate. I think it will be important to document the escalation path here, but it doesn't have to be this way.

The elephant in the room is: who has the authority to decide about the checks that new code has to pass?

Reply:
hack/ OWNERS, who will also be approving the script itself and other similar lints?

Reply:
That looks like a set of individuals, not any particular SIG. That's fine with me, I was really just wondering whether there is a SIG which is responsible for code quality questions.

Should `.golangci-strict.yaml` then be under `hack`, too? `.golangci.yaml` was placed in the root because that is where `golangci-lint` found it when walking up the stack, but the script doesn't rely on that anymore.

Reply:
Oh I see! I think code organization is the closest thing at the moment as a specific subset of architecture, but typically tooling for code quality would be sig testing or more narrow like the logging linter.

I am a proponent of not dumping everything in the repo root both for ease of approvals and for making things less cluttered to discover ... The only counterpoint I can think of is if people try to run outside of our scripts, but that already doesn't make much sense for the non-default config. (It also probably doesn't work well anyhow for the other one)

Reply:
Fully agreed. It also doesn't need to be a dot file anymore... will change it.

Reply:
IIRC we had to add it to the top because there was to be additional executions of the `golangci-lint` process so we can lint the modules too

Reply:
When using the `--config` parameter, the file can be anywhere.

So perhaps I should also move `.golangci.yaml`, unless someone can think of a reason why it needs to be in the root?

Reply:
I don't see any problem if it works for the vendored modules, @thockin played with it a bit more

Reply:
The file gets moved as part of https://github.com/kubernetes/kubernetes/pull/116166.